### DISCUSSION TITLE
Improving Performance Issues in Saleor Due to Token Expiration

### DISCUSSION THREAD

**Discussion by Ashiqquee** [saleor/saleor#16017] [Q&A]
Hi all,

I am having some issues with Saleor. When I am developing an app using Saleor, the main issue is that when I make an API call a second time with the token, it shows "signature expired." Due to this, the performance is very poor.

**Comment by timuric** [comment]
you should be using app tokens (configurations->apps) instead of admin token https://docs.saleor.io/docs/3.x/api-usage/authentication
